avocado-framework/avocado
# Requirements database [v3]

This PR adds the ability to spawners to store requirements into cache.
This will make more efficient running multiple tests with the same
requirement or rerunning the jobs itself.

This also brings ability to use requirements with podman spawner. So now
is possible to run something like this:
```
class SuccessTest(Test):

    def check_hello(self):
        result = process.run("hello", ignore_status=True)
        self.assertEqual(result.exit_status, 0)
        self.assertIn('Hello, world!', result.stdout_text,)

    def test_a(self):
        """
        avocado dependency={"type": "package", "name": "hello"}
        """
        self.check_hello()
```

Warning: Right now, the requirement cache doesn't have any controls for
clearing cache or updating, and also didn't check the current
environment. So if the user do some changes to the environment outside
Avocado runtime, like uninstalling package, podman prune, removing
asset, etc. The tests with requirements will be broken.

Signed-off-by: Jan Richter <jarichte@redhat.com>

---

Changes from v2 (#5418):
* Use UPPER_CASE on enum

Changes from v1 (#5412):

* Typo fixes.
* Usage of  list comprehensions.
* New and well-defined `RuntimeTask` states.
* Usage of hostname in cache for process spawner.
* Change `test_requirements.py` to avocado `Test` and usage of varianter.
* Documentation update.
